# Step 1
Extract the first two columns, which correspond to the proteins, as well as the last two columns, which contain the e-values and scores, from the file generated by the homology comparison between the two species.
```python
#Input
import pandas as pd
df = pd.read_csv("/home/liuhuiling/gene_protein/Bter_Dros/Dros_Bter_protein.txt", delimiter="\t", header=None)
df_extracted = df.iloc[:, :2].copy() 
df_extracted[2] = df.iloc[:, -2] 
df_extracted[3] = df.iloc[:, -1]
df_extracted.columns = ["Dros", "Bter", "E value", "score"]
df_extracted.to_csv("/home/liuhuiling/gene_protein/Bter_Dros/AM_Dros protein comparison.csv", index=False)
```
# Step 2
Extract the corresponding gene and protein annotations from the GFF files
```R
gff_file <- "/home/liuhuiling/gene_protein/GFF/Apis_mellifera_genomic.gff"
parse_gff <- function(gff_file) {
  gene_protein_mapping <- list()
  # Reading GFF Files
  gff_data <- read.delim(gff_file, sep="\t", header=FALSE, stringsAsFactors=FALSE,col.names=c("seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attributes"))
  # Remove comment lines
  gff_data <- gff_data[!startsWith(gff_data$seqname, "#"), ]
  # Filter rows for CDS regions
  cds_rows <- gff_data[gff_data$feature %in% c("CDS", "gene"), ]
  # Extracting the correspondence between genes and proteins
  for (i in 1:nrow(cds_rows)) {
    attributes <- unlist(strsplit(cds_rows$attributes[i], ";"))
    gene_name <- NULL
    protein_name <- NULL
    
    for (attr in attributes) {
      if (startsWith(attr, "gene=")) {
        gene_name <- gsub("gene=", "", attr)
      } else if (startsWith(attr, "protein_id=")) {
        protein_name <- gsub("protein_id=", "", attr)
      }
    }
    
    if (!is.null(gene_name) && !is.null(protein_name)) {
      gene_protein_mapping[[protein_name]] <- gene_name
    }
  }
  
  return(gene_protein_mapping)
}

# Call the function to extract the correspondence between proteins and genes
gene_protein_mapping <- parse_gff(gff_file)

# Output
gene_protein_df <- data.frame(AM_Protein = names(gene_protein_mapping),AM_Gene = unlist(gene_protein_mapping))
write.csv(gene_protein_df, "/home/liuhuiling/gene_protein/AM_protein_gene_mapping.csv", row.names=FALSE)
```
# Step 3
Merge the files `AM_Dros protein comparison.csv`, `AM_protein_gene_mapping.csv`, and `Dros_protein_gene_mapping.csv` to generate a list of homologous genes between *Drosophila melanogaster* and *Apis mellifera*.
```python
#Input
import pandas as pd
protein_comparison_df = pd.read_csv("/home/liuhuiling/gene_protein/AM_Dros/AM_Dros protein comparison.csv")
AM_protein_gene_mapping_df = pd.read_csv("/home/liuhuiling/gene_protein/AM_protein_gene_mapping.csv")
Dros_protein_gene_mapping_df = pd.read_csv("/home/liuhuiling/gene_protein/Dros_protein_gene_mapping.csv")

#merge data
result_df = pd.merge(protein_comparison_df, AM_protein_gene_mapping_df, left_on="AM", right_on="AM_Protein", how="left")
result_df = pd.merge(result_df, Dros_protein_gene_mapping_df, left_on="Dros", right_on="Dros_Protein", how="left")
print(result_df.columns)
result_df.drop(columns=["AM_Protein", "Dros_Protein"], inplace=True)
result_df.dropna(inplace=True)
result_df.sort_values(by=["E value", "score"], ascending=[True, False], inplace=True)
result_df.drop_duplicates(subset=["AM_Gene"], keep="first", inplace=True)
result_df.drop_duplicates(subset=["Dros_Gene"], keep="first", inplace=True)

# Output
result_df.to_csv("/home/liuhuiling/gene_protein/AM_Dros/AM_Dros gene result.csv", index=False)
```
# Step 4
## For the *Drosophila melanogaster* dataset
We obtained the metadata and gene expression count matrices directly from the associated publication and performed minor formatting adjustments to prepare the data for downstream analysis.
### Gene expression count matrices
```R
#input count
import pandas as pd
df = pd.read_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.midgut.umi.geneMatrix.txt", delimiter="\t", header=0)
print(df.head(10))
df.rename(columns={df.columns[0]: "Dros_gene_id"}, inplace=True)
print(df.head(10))
df = df.applymap(lambda x: str(x).replace('"', ''))

#Output
df.to_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.midgut.umi.geneMatrix.txt", sep="\t", header=True,index=False)
```
### Dros.pheno.cell.txt
The first row of the expression matrix, containing the cell names, was extracted and saved as a separate column in a file. This step was performed to align the order of cells in the metadata with that in the expression matrix for subsequent analyses.
```python
#Extracte cell information from expression matrix
#Input
import pandas as pd
df = pd.read_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.midgut.umi.geneMatrix.txt", delimiter="\t", header=None)
print(df.head(10))
new_column = df.iloc[0, :].reset_index(drop=True)
new_column.name = 'Sample_ID'
#Output
new_column.to_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.pheno.cell.txt", sep="\t", index=False)

#Extract cell and group information from metadata
#Input metadata
import pandas as pd
df1=pd.read_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.midgut.metadata.txt", delimiter=",", header=0)
print(df1.head(10))
df1 = df1.applymap(lambda x: str(x).replace('"', ''))
selected_columns = df1[["cell", "celltype"]]
selected_columns.to_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/selected_columns.txt", sep="\t", index=False)
#Output
df1.to_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.midgut.metadata.txt", sep="\t", index=False)

#These files were merged to generate the final Dros.pheno.cell.txt file.
#Input
import os
os.environ.clear()
import pandas as pd
Dros_pheno_cell=pd.read_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.pheno.cell.txt", delimiter="\t", header=0)
selected_columns=pd.read_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/selected_columns.txt", delimiter="\t", header=0)
result_df = pd.merge(Dros_pheno_cell, selected_columns, left_on="Sample_ID", right_on="cell", how="left")
print(result_df.head(10))
result_df.drop(columns=["cell"], inplace=True)
result_df.columns.values[1] = "Celltype"
result_df.insert(1, "Study_ID", "Dros")
result_df.to_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.pheno.cell.txt", sep="\t", index=False)
selected_columns = df[["Sample_ID", "Study_ID","Celltype"]].copy()
selected_columns["Celltype"] = "Dros_" + selected_columns["Celltype"].astype(str)
#Output
selected_columns.to_csv("/home/liuhuiling/gut_snRNA/cross_sp_code/test/Dros.pheno.cell.txt", sep="\t", index=False)
```
## For the *Apis mellifera* database
We extracted cell ordering, metadata, and gene expression count matrices from the Seurat .RDS file to enable downstream integration and analysis.

